src/main/src/main/java/org/geoserver/function/IsInstanceOfFactory.java

```diff
@@ -0,0 +1,47 @@
+package org.geoserver.function;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.geoserver.catalog.Predicates;
+import org.geotools.feature.NameImpl;
+import org.geotools.filter.FunctionFactory;
+import org.opengis.feature.type.Name;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Function;
+import org.opengis.filter.expression.Literal;
+
+/**
+ * This class implements the {@link FunctionFactory} interface and can be used for creating a new {@link IsInstanceOf} function from the input
+ * parameters.
+ * 
+ * @author Nicola Lagomarsini geosolutions
+ */
+public class IsInstanceOfFactory implements FunctionFactory {
+
```

Comment:
Isn't there any other FunctionFactory in GeoServer that we can add this FunctionFactory functions to?
If there is none, can we imagine to rename this to something more general and become a factory for more functions in the future?
I would avoid to have a factory for each single Function we need.
Something like CatalogFunctionFactory, or ConfigFunctionFactory, maybe?